ironfish/src/workerPool/tasks/decryptNotes.ts

```diff
@@ -237,7 +238,7 @@ export class DecryptNotesTask extends WorkerTask {
           forSpender: false,
           hash: note.hash(),
           nullifier:
-            currentNoteIndex !== null
+            currentNoteIndex !== null && spendingKey
```

Comment:
Is it possible to add a spendingKey to an account?

The account would need to rescan in order to get the `decryptedNotes` and `nullifierToNoteHash` stores filled correctly

Reply:
Currently it is not supported. I don't think it would be worth it to support, instead the user could just delete the account and reimport using spending key. That way all the existing machinery just takes care of the process.